synccache/auto_refresh_cache.go

```diff
@@ -133,16 +123,56 @@ func (w *autoRefreshCache) sync(ctx context.Context) {
 		// If not ok, it means evicted between the item was evicted between getting the keys and this update loop
 		// which is fine, we can just ignore.
 		if value, ok := w.lruMap.Peek(k); ok {
-			newItem, result, err := w.syncCb(ctx, value.(CacheItem))
+			wrapper := value.(cacheItemWrapper)
+			if wrapper.RetryCount >= w.maxRetries {
```

Comment:
maybe make this >?  Otherwise, if someone specified a max retry of 0, it'll never get synced.

Reply:
```suggestion
			if wrapper.RetryCount > w.maxRetries {
```